package.json

```
@@ -130,6 +130,14 @@
     "webpack": "^5.44.0",
     "webpack-dev-server": "^3.11.2"
   },
+  "peerDependencies": {
+    "typescript": ">=4.2.0 < 5"
```

I've recently learned that TS doesn't follow semver. There may, and most likely will be breaking changes between 4.2 and 4.3. We should look the supported version by the minor range:
```suggestion
    "typescript": ">=4.2.0 < 4.3.0"
```

Please, would you add this change?

---

So you had originally said:

> Please use the newer version of TypeScript (the latest is 4.4, the minimum required should be 4.2).

So I'm not sure how to reconcile that with this.

---

perhaps what you want is `>= 4.2.x <= 4.4.x` (I'll have to check if that's valid syntax

---

I've said that initially before learning that TypeScript [ships breaking changes between minor releases](https://github.com/microsoft/TypeScript/issues/14116#issuecomment-280592571). With that knowledge, I suggest we know lock the peer dependency version to exact major + minor versions, allowing only the latest patches.

---

Ok, it's just typescript is up to 4.5.2, but if you are only comfortable saying it works for 4.2.x, it's your lib, up to you of course.

---

We can bump the supported version to 4.5.2 or anything else once we have the guarantee the library works with that version. That means switching to that version in dev dependencies and using it locally. I've had quite a number of painful minor version updates in TS so I highly recommend we don't take minor versions compatibility as a given. 

Going with `4.2.x` is the correct approach at the moment 👍